qiskit_nature/drivers/base_driver.py

```diff
@@ -50,6 +50,13 @@ def __init__(self,
         self._hf_method = hf_method
         self._supports_molecule = supports_molecule
 
+    @abstractmethod
```

Comment:
I imagine the reason we never put a run() method here before is that the type returned from the two subclasses is different and hence we left the run methods to be specific for the fermionic and bosonic drivers but did not try to have some common parent. If this is needed for some reason then maybe it should return 'Any' ?

Reply:
Yes, this is indeed the case. I introduced a BaseProblem class and it makes sense that it operates on a BaseDriver (it happens that it also runs a driver). I could try 'Any'. We also discussed with Max that these classes will likely be refactored later somehow and we will try to come up with a better solution.